### ISSUE TITLE
Be less strict about restoring packages installed from sources from CRAN

### ISSUE THREAD

**Issue by kevinushey** [rstudio/packrat#371]
Lots of users run into issues where a package installed from a local source tarball later cannot be snapshotted / restored, with something like:

> Error: Unable to retrieve package records for the following packages: 
> - 'dplyr'

Currently, Packrat only attempts to install packages from CRAN if it sees within the DESCRIPTION file

    Repository: CRAN

and while the main CRAN repositories do set this for the packages they provide (with a couple exceptions for e.g. 'extra' repositories), many users set up their own CRAN-like repositories that do not enforce this.

In short, we should consider being more permissive with packages installed from source tarballs, and allow those packages to be reinstalled from CRAN if an appropriate version is available.

**Comment by slopp** [comment]
@kevinushey - can we escalate this? We had another customer (Bank of America) who ran into this issue after a custom install of the stringi package.

**Comment by colearendt** [comment]
One test related to the lack of a "Repository: " field included in PR #369 .  Focus here is local repository, but perhaps relevant in some capacity.  At present the test has been `skip()`ed.
